torch/csrc/jit/script/module.h

```diff
@@ -563,6 +563,14 @@ struct Module {
     }
   }
 
+  void set_metadata(const std::string& metadata) {
```

Comment:
This API is not symmetric: on save metadata is taken from an argument, but on load it is made to be part of the module. This is weird: meta-data is stored on the model but it is getting stored on a Module on load. Then a bunch of submodule will have meta-data that is unused. This is a situation that we wanted to avoid by putting meta-data at the module level rather than on each module. Can the load function also just gain a std::string* that will be where it puts metadata if found?